specutils/io/default_loaders/tabular_fits.py

```diff
@@ -71,8 +74,13 @@ def tabular_fits_loader(file_obj, column_mapping=None, hdu=1, **kwargs):
     # routines to search for spectral axis information in the file.
     with read_fileobj_or_hdulist(file_obj, **kwargs) as hdulist:
         wcs = WCS(hdulist[hdu].header)
+        primary_header = hdulist[0].header
+        tab = Table.read(hdulist[hdu])
 
-    tab = Table.read(file_obj, format='fits', hdu=hdu)
+    if store_data_header:
+        tab.meta = hdulist[hdu].header
```

Comment:
I am quite puzzled how this is working at all, accessing `hdulist` outside of its context (but apparently it does at least in my local installation). Wondering then why reading `hdulist[0].header` would not work without the temporary variable...
In principle there should not even be another read required, since this is what `Table` sets by default, but its reader is converting `meta` to `dict`, which is arguably less convenient than the `fits.Header`.
Highlights that the new option is not tested yet, as in fact there are no tests on the `meta['header']` returned on the _reader_ side.

Reply:
You know, I'm confused how it's working too 😅 . I'm going to move it into the context to at least make me feel better.